src/tools/rbd/action/MirrorPool.cc

```diff
@@ -1414,50 +1448,45 @@ int execute_status(const po::variables_map &vm,
 
   librbd::RBD rbd;
 
-  std::map<librbd::mirror_image_status_state_t, int> states;
-  r = rbd.mirror_image_status_summary(io_ctx, &states);
+  uint32_t total_images = 0;
+  std::map<librbd::mirror_image_status_state_t, int> mirror_image_states;
+  MirrorHealth mirror_image_health = MIRROR_HEALTH_UNKNOWN;
+  r = get_mirror_image_status(io_ctx, &total_images, &mirror_image_states,
+                              &mirror_image_health);
   if (r < 0) {
-    std::cerr << "rbd: failed to get status summary for mirrored images: "
-	      << cpp_strerror(r) << std::endl;
     return r;
   }
 
-  if (formatter != nullptr) {
-    formatter->open_object_section("status");
+  MirrorDaemonServiceInfo daemon_service_info(io_ctx);
+  r = daemon_service_info.init();
+  if (r < 0) {
+    return r;
   }
 
-  enum Health {Ok = 0, Warning = 1, Error = 2} health = Ok;
-  const char *names[] = {"OK", "WARNING", "ERROR"};
-  int total = 0;
+  MirrorHealth mirror_daemon_health = daemon_service_info.get_daemon_health();
+  auto mirror_services = daemon_service_info.get_mirror_services();
 
-  for (auto &it : states) {
-    auto &state = it.first;
-    if (health < Warning &&
-	(state != MIRROR_IMAGE_STATUS_STATE_REPLAYING &&
-	 state != MIRROR_IMAGE_STATUS_STATE_STOPPED)) {
-      health = Warning;
-    }
-    if (health < Error &&
-	state == MIRROR_IMAGE_STATUS_STATE_ERROR) {
-      health = Error;
-    }
-    total += it.second;
-  }
+  auto mirror_health = std::max(mirror_image_health, mirror_daemon_health);
 
   if (formatter != nullptr) {
+    formatter->open_object_section("status");
     formatter->open_object_section("summary");
-    formatter->dump_string("health", names[health]);
+    formatter->dump_stream("health") << mirror_health;
+    formatter->dump_stream("daemon_health") << mirror_daemon_health;
+    formatter->dump_stream("pool_health") << mirror_image_health;
     formatter->open_object_section("states");
-    for (auto &it : states) {
+    for (auto &it : mirror_image_states) {
       std::string state_name = utils::mirror_image_status_state(it.first);
       formatter->dump_int(state_name.c_str(), it.second);
     }
     formatter->close_section(); // states
     formatter->close_section(); // summary
   } else {
-    std::cout << "health: " << names[health] << std::endl;
-    std::cout << "images: " << total << " total" << std::endl;
-    for (auto &it : states) {
+    std::cout << "overall health: " << mirror_health << std::endl;
```

Comment:
@dillaman After this change need to update `wait_for_pool_healthy` in 
qa/workunits/rbd/rbd_mirror_stress.sh.  See e.g. this stress test failure [1].

[1]  http://qa-proxy.ceph.com/teuthology/trociny-2019-12-04_08:44:42-rbd-wip-mgolub-testing-distro-basic-smithi/4565918/teuthology.log

Reply:
Thx -- tweaked